docs/design/asset-canister-interface.md

```diff
@@ -0,0 +1,508 @@
+# Asset Canister Interface
```

Comment:
First off, I think it's nice that you are separating these concepts: Asset Canister Interface and Asset Canister implementation. For me, this is helpful in understanding the things I can and can't do as a developer.

Maybe we also ought to think about renaming our asset canister implementation so it's clear its _one_ possible implementation of many of the interface.

Reply:
It may also be useful (predominantly to veteran IC developers) to note that this is not to be confused with the _implementation_ of the asset canister that they are likely accustomed to using

Reply:
Before today's commit https://github.com/dfinity/sdk/pull/3258/commits/43660ece47261bab5a136f577bdaec0fe262d66f I had this:

> This document is meant to describe the interface, not the idiosyncrasies of the implementation.

I wonder if I should re-add something like this